### PR TITLE
Hide sensitive information by config

### PR DESCRIPTION
**Pull request by erikn69**
#1452 shows a security problem on debugbar files, this PR offer posibility of hide data by config, sometimes can be used not default keys like `pass`, `token`, or others
Only `request_request.password` is hidden from `request_request`
https://github.com/barryvdh/laravel-debugbar/blob/27b088aaad2adba74a7a73a47cb8d11277b551e3/src/DataCollector/RequestCollector.php#L128-L130

**CC:** @barryvdh 

**UPDATE:** Now it hide keys on clockwork too, also it hide `authorization` token(security fix)

### CODE REVIEW

**Comment by oprudkyi** [comment]
Hi @barryvdh 
could you please take a look at this PR
it would be great to have configurable list of hidden keys in some scenarios (multiple qa, non-standard keys)

**Comment by oprudkyi** [comment]
Hi @barryvdh

is anything can be done here to move it forward ?

File: src/DataCollector/SessionCollector.php [hunk 1]
```diff
@@ -30,6 +35,13 @@ public function collect()
         foreach ($this->session->all() as $key => $value) {
             $data[$key] = is_string($value) ? $value : $this->formatVar($value);
         }
+
```
Comment (oprudkyi) [inline src/DataCollector/SessionCollector.php:38]: Hi @erikn69 
this won't handle deep array values as they already converted to string, probably all collect() should look like 
```
    /**
     * {@inheritdoc}
     */
    public function collect()
    {
        $data = $this->session->all();

        foreach ($this->hiddens as $key) {
            if (Arr::has($data, $key)) {
                Arr::set($data, $key, '******');
            }
        }

        foreach ($data as $key => $value) {
            $data[$key] = is_string($value) ? $value : $this->formatVar($value);
        }


        return $data;
    }

```
i.e. apply hiddens before converting values to strings

**Comment by oprudkyi** [comment]
Hi @erikn69 
also, probably worth to add the same support into src/Support/Clockwork/ClockworkCollector.php 
it includes the same data - cookies, headers etc

**Comment by erikn69** [comment]
> also, probably worth to add the same support into src/Support/Clockwork/ClockworkCollector.php

@oprudkyi you are right, but I don't know if this PR is going to be merged, anyway I'm going to try

**Review by barryvdh — APPROVED** [review]

**Pull request merged**